### PR TITLE
Improve validation

### PR DESCRIPTION
**Pull request by makyo**
* Loosen relation validation
* Normalize machines to ints
* Allow 'new' machines

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/24/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/25/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/26/
Test FAILed.

File: jujubundlelib/models.py [hunk 1]
```diff
@@ -103,3 +103,15 @@ def parse_v4_unit_placement(placement_str):
...
     return UnitPlacement(container, machine, service, unit)
+
+
+def normalize_machines(machines):
+    """Normalize the machines spec to use integer keys."""
+    normalized_machines = {}
+    try:
+        for k, v in machines.items():
+            normalized_machines[int(k)] = v
+    except (AttributeError, ValueError):
```
Comment (frankban) [inline jujubundlelib/models.py]: We should also catch the TypeError raised in the case k is None (and add a test).

**Comment by frankban** [comment]
:+1: with one request.
Thank you!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/27/
Test PASSed.

**Comment by makyo** [comment]
Thanks! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://localhost:8080/job/juju-bundlelib-merge

**Pull request merged**